Model/WebhookProcessor.php

```diff
@@ -187,18 +203,26 @@ public function enqueueWebhook($params, $name)
     }
 
     /**
-     * @param $params
-     * @param $webhook
+     * ProcessWebhookCron
+     *
+     * @param array $params
+     * @param mixed $webhook
      * @throws LocalizedException
-     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws NoSuchEntityException
      */
     public function processWebhookCron($params, $webhook)
     {
         $isTransactionSuccess = false;
         $order = $this->orderFactory->create()->loadByIncrementId((string)$params['externalReferenceId']);
+        $this->updateWebhookQueue($webhook->getEntityId(), 'status', self::IN_PROGRESS);
         if (!$order || !$order->getId()) {
-            $attempts = $webhook->getAttempts() + 1;
-            $this->updateAttempts($webhook->getEntityId(), 'attempts', $attempts);
+            if ($webhook->getAttempts() >= 3) {
+                $this->updateWebhookQueue($webhook->getEntityId(), 'status', self::FAILED);
+            } else {
+                $this->updateWebhookQueue($webhook->getEntityId(), 'status', self::PENDING);
+                $attempts = $webhook->getAttempts() + 1;
+                $this->updateWebhookQueue($webhook->getEntityId(), 'attempts', $attempts);
+            }
```

Comment:
As we discussed, let's have the Webhook Cron (should be renamed to BalanceQueue) responsible for managing the queue and the job status. WebhookProcessor should not be aware of the queue.